### PR TITLE
[doctest] update `ops` module documentation

### PR DESCRIPTION
**Pull request by andrijapau**
Follow up to https://github.com/PennyLaneAI/pennylane/pull/8329.

[sc-100039]

### CODE REVIEW

File: pennylane/ops/functions/iterative_qpe.py [hunk 1]
```diff
@@ -56,26 +56,23 @@ def circuit():
...
-
-        >>> print(circuit())
-        [[0 0 1]
-         [0 0 1]
-         [0 0 1]
-         [1 1 1]
-         [0 0 1]]
+    >>> result = circuit()
+    >>> assert result.shape == (5, 3)
+    >>> print(result) # doctest: +SKIP
```
Comment (albi3ro) [inline pennylane/ops/functions/iterative_qpe.py]: Maybe set a seed on the device?

File: pennylane/ops/functions/map_wires.py [hunk 2]
```diff
@@ -62,13 +62,16 @@ def circuit():
...
+            qml.capture.disable()
+
         >>> jaxpr
         { lambda ; . let
-            _:AbstractOperator() = Hadamard[n_wires=1] 1
-            a:AbstractOperator() = PauliZ[n_wires=1] 1
-            b:AbstractMeasurement(n_wires=None) = expval_obs a
-        in (b,) }
+          _:AbstractOperator() = Hadamard[n_wires=1] 1:i...[]
+          a:AbstractOperator() = PauliZ[n_wires=1] 1:i...[]
```
Comment (albi3ro) [inline pennylane/ops/functions/map_wires.py]: Perhaps we should run this with a fixed x64 mode?

File: pennylane/ops/meta.py [hunk 3]
```diff
@@ -200,15 +200,7 @@ def circuit():
             return qml.expval(qml.X(0))
 
     >>> qml.snapshots(circuit)()
```
Comment (albi3ro) [inline pennylane/ops/meta.py]: perhaps using pprint on this?

File: pennylane/ops/op_math/composite.py [hunk 4]
```diff
@@ -337,9 +337,9 @@ def label(self, decimals=None, base_label=None, cache=None):
 
         >>> op = qml.S(0) + qml.X(0) + qml.Rot(1,2,3, wires=[1])
         >>> op.label()
-        '(S+X)+Rot'
+        '𝓗'
         >>> op.label(decimals=2, base_label=[["my_s", "my_x"], "inc_rot"])
-        '(my_s+my_x)+inc_rot\n(1.00,\n2.00,\n3.00)'
+        "[['my_s', 'my_x'], 'inc_rot']\n(1.00,\n2.00,\n3.00)"
```
Comment (albi3ro) [inline pennylane/ops/op_math/composite.py]: Perhaps worth getting rid of this example now, since it looks like we don't have the same ability to handle a list of base labels anymore.

File: pennylane/ops/op_math/pow.py [hunk 5]
```diff
@@ -146,16 +148,13 @@ def __new__(cls, base=None, z=1, id=None):
         different based on ``base``'s inheritance.  We cache the different types in private class
         variables so that:
 
+        >>> z = 2
         >>> Pow(op, z).__class__ is Pow(op, z).__class__
         True
         >>> type(Pow(op, z)) == type(Pow(op, z))
         True
         >>> isinstance(Pow(op, z), type(Pow(op, z)))
         True
```
Comment (albi3ro) [inline pennylane/ops/op_math/pow.py:157]:
```suggestion
        True
        >>> Pow(qml.RX(1.2, wires=0), 0.5).__class__ is PowOperation
        True
```

File: pennylane/ops/op_math/sprod.py [hunk 6]
```diff
@@ -57,10 +57,10 @@ def s_prod(scalar, operator, lazy=True, id=None):
         >>> qml.matrix(op).shape
         (3, 2, 2)
 
-        But it doesn't support batching of operators:
+        It also supports batching of operators:
```
Comment (albi3ro) [inline pennylane/ops/op_math/sprod.py]: Wait what? When did that happen?

Comment (albi3ro) [reply pennylane/ops/op_math/sprod.py]: How does that even work?

File: pennylane/ops/op_math/composite.py (hunk 4 above)
Comment (andrijapau) [reply pennylane/ops/op_math/composite.py]: Done in https://github.com/PennyLaneAI/pennylane/pull/8330/commits/50f4c2ca711c3736d72cd6b9bb2dafeb1f30f7de

File: pennylane/ops/op_math/sprod.py (hunk 6 above)
Comment (albi3ro) [reply pennylane/ops/op_math/sprod.py]: Pretty sure we don't support this, but we just don't *immediately* error out on initialization.

File: pennylane/ops/meta.py (hunk 3 above)
Comment (andrijapau) [reply pennylane/ops/meta.py]: Done in https://github.com/PennyLaneAI/pennylane/pull/8330/commits/c3f199f2ebd8351d1b83b882c1b947aeb95a7e69

File: pennylane/ops/op_math/sprod.py (hunk 6 above)
Comment (andrijapau) [reply pennylane/ops/op_math/sprod.py:60]: 🤷🏼 Didn't look into it, seemed like a good thing?

File: pennylane/ops/functions/map_wires.py (hunk 2 above)
Comment (andrijapau) [reply pennylane/ops/functions/map_wires.py]: Done in https://github.com/PennyLaneAI/pennylane/pull/8330/commits/79a6cbe0a06a859d5acb155d570d1380b4e89549

File: pennylane/ops/functions/iterative_qpe.py (hunk 1 above)
Comment (andrijapau) [reply pennylane/ops/functions/iterative_qpe.py]: Done in https://github.com/PennyLaneAI/pennylane/pull/8330/commits/8032fbe07a4ffb5c6419ce031bbed4d6b2919545

File: pennylane/ops/op_math/sprod.py (hunk 6 above)
Comment (andrijapau) [reply pennylane/ops/op_math/sprod.py]: Hmm, I see.

Comment (andrijapau) [reply pennylane/ops/op_math/sprod.py]: Yeah looking at the output now, I don't know how it would work lol

Comment (andrijapau) [reply pennylane/ops/op_math/sprod.py]: We should probably have it error out sooner no?

Comment (albi3ro) [reply pennylane/ops/op_math/sprod.py]: I might just delete this block entirely. We give no indication anywhere that an operator could be replaced by a list of operators.

File: pennylane/ops/op_math/sum.py [hunk 7]
```diff
@@ -201,9 +206,10 @@ def circuit(weights):
                 qml.RX(weights[2], wires=1)
                 return qml.expval(sum_op)
 
+        >>> import pennylane.numpy as qnp
         >>> weights = qnp.array([0.1, 0.2, 0.3], requires_grad=True)
```
Comment (albi3ro) [inline pennylane/ops/op_math/sum.py]:
```suggestion
        >>> import pennylane.numpy as pnp
        >>> weights = pnp.array([0.1, 0.2, 0.3], requires_grad=True)
```

File: pennylane/ops/meta.py [hunk 8]
```diff
@@ -199,16 +199,13 @@ def circuit():
...
-            [1, 1],
-            [1, 1],
-            [1, 1]]),
-    'execution_results': 0.0}
+    >>> from pprint import pprint
+    >>> pprint(qml.snapshots(circuit)())
+    {0: np.float64(1.0),
+     2: array([0.70710678+0.j, 0.        +0.j, 0.        +0.j, 0.70710678+0.j]),
+     'execution_results': np.float64(0.0),
+     'samples': ...,
```
Comment (albi3ro) [inline pennylane/ops/meta.py]: was this being stochastic?

Comment (andrijapau) [reply pennylane/ops/meta.py]: Yes

Comment (andrijapau) [reply pennylane/ops/meta.py]: I can seed the device though

File: pennylane/ops/op_math/sprod.py (hunk 6 above)
Comment (andrijapau) [reply pennylane/ops/op_math/sprod.py]: https://github.com/PennyLaneAI/pennylane/pull/8330/commits/a73ed1471bfebcd1674bccb2200bdf35d4c2e505

File: pennylane/ops/meta.py (hunk 8 above)
Comment (andrijapau) [reply pennylane/ops/meta.py]: https://github.com/PennyLaneAI/pennylane/pull/8330/commits/c74195ecf0d30c85f412114c21e31a7ab819b65f

**Review by albi3ro — APPROVED** [review]

File: .github/workflows/documentation-tests.yml [hunk 9]
```diff
@@ -64,7 +64,8 @@ jobs:
           --ignore=pennylane/math
           --ignore=pennylane/compiler
           --ignore=pennylane/measurements
-          --ignore=pennylane/ops
+          --ignore=pennylane/ops/qubit
+          --ignore=pennylane/ops/qutrit
```
Comment (andrijapau) [inline .github/workflows/documentation-tests.yml:67]: These are annoying and will have to be left for another PR.

File: pennylane/ops/functions/matrix.py [hunk 10]
```diff
@@ -135,55 +132,21 @@ def circuit(theta):
...
 
-        >>> theta = np.array(0.3, requires_grad=True)
+        >>> theta = pnp.array(0.3, requires_grad=True)
+        >>> # Expected value is 2 * cos(0.3 / 2)
         >>> cost(theta)
-        1.9775421558720845
-        >>> qml.grad(cost)(theta)
-        -0.14943813247359922
-
-        .. note::
```
Comment (gabrielasd) [inline pennylane/ops/functions/matrix.py:154]: Why did L154-L186 get removed?

Comment (andrijapau) [reply pennylane/ops/functions/matrix.py:154]: Good question, we don't infer wires anymore. This documentation section is super old lol.

Comment (gabrielasd) [reply pennylane/ops/functions/matrix.py:154]: Got it! Thanks for the context :)
